### PR TITLE
FPO-381: Randomly sample and contextualise descriptions

### PR DESCRIPTION
**Pull request by willfish**
### Jira link

FPO-323

### What?

I have added/removed/altered:

- Adds ability to sample and contextualise search results
- Contextualise goods nomenclature hierarchy

### Why?

I am doing this because:

- This is required to randomly sample and present real searches

### AC

- Tested
- Shared with stakeholders
- Pair reviewed

### CODE REVIEW

File: src/models/goodsNomenclature.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+    allAncestorDescriptions = allAncestorDescriptions.concat(ancestorDescriptions)
+    allAncestorDescriptions = allAncestorDescriptions.filter((description: string | null) => description !== null)
+
+    goodsNomenclature.sampleDescription = sampleDescription
+    goodsNomenclature.goods_nomenclature_item_id = attributes.goods_nomenclature_item_id
+    goodsNomenclature.description = attributes.description
+    goodsNomenclature.formattedDescription = attributes.formatted_description
+    goodsNomenclature.ancestorDescriptions = allAncestorDescriptions
+    console.debug('Goods Nomenclature')
```
Comment (alexdesi) [inline src/models/goodsNomenclature.ts:42]: why all these debug?
should those be `console.log`?

File: src/models/goodsNomenclature.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+  static getSectionDescription (response: any): string {
+    const relation = response.data.relationships.section.data
+    const included = GoodsNomenclature.lookupIncluded(response, relation)
+
+    console.debug('Section')
+    console.debug(included)
+
+    return included.attributes.title
+  }
```
Comment (alexdesi) [inline src/models/goodsNomenclature.ts:55]: there is no error handling here,
and response can be any,
sure anything will go as expected?

**Review by alexdesi — APPROVED** [review]
I gave a fast look,
it looks good.

File: src/models/goodsNomenclature.ts (hunk 2 above)
Comment (willfish) [reply src/models/goodsNomenclature.ts:47]: I could validate the schema of the response if it goes wrong for some reason but I'm kind of taking care of that by validating the result status codes and would have failed much higher than this.

File: src/models/goodsNomenclature.ts (hunk 1 above)
Comment (willfish) [reply src/models/goodsNomenclature.ts]: Removed

**Pull request merged**